src/main/java/eu/europeana/metis/sandbox/service/workflow/HarvestPublishServiceImpl.java

```diff
@@ -32,46 +33,48 @@ public HarvestPublishServiceImpl(HarvestService harvestService,
     }
 
     @Override
-    public CompletableFuture<Void> runHarvestZipAsync(MultipartFile file, DatasetMetadata datasetMetadata){
+    public CompletableFuture<Void> runHarvestFileAsync(MultipartFile file, DatasetMetadata datasetMetadata,
+                                                       CompressedFileExtension compressedFileExtension) {
         try {
             Record.RecordBuilder recordDataEncapsulated = Record.builder()
-                                                                .datasetId(datasetMetadata.getDatasetId())
-                                                                .datasetName(datasetMetadata.getDatasetName())
-                                                                .country(datasetMetadata.getCountry())
-                                                                .language(datasetMetadata.getLanguage());
-            return runHarvestZipAsync(file.getInputStream(), recordDataEncapsulated, datasetMetadata);
+                    .datasetId(datasetMetadata.getDatasetId())
+                    .datasetName(datasetMetadata.getDatasetName())
+                    .country(datasetMetadata.getCountry())
+                    .language(datasetMetadata.getLanguage());
+            return runHarvestFileAsync(file.getInputStream(), recordDataEncapsulated, datasetMetadata, compressedFileExtension);
         } catch (IOException e) {
             throw new ServiceException("Error harvesting records from file " + file.getName(), e);
         }
     }
 
     @Override
-    public CompletableFuture<Void> runHarvestHttpZipAsync(String url, DatasetMetadata datasetMetadata){
+    public CompletableFuture<Void> runHarvestHttpZipAsync(String url, DatasetMetadata datasetMetadata) {
         Record.RecordBuilder recordDataEncapsulated = Record.builder()
-                                                            .datasetId(datasetMetadata.getDatasetId())
-                                                            .datasetName(datasetMetadata.getDatasetName())
-                                                            .country(datasetMetadata.getCountry())
-                                                            .language(datasetMetadata.getLanguage());
+                .datasetId(datasetMetadata.getDatasetId())
+                .datasetName(datasetMetadata.getDatasetName())
+                .country(datasetMetadata.getCountry())
+                .language(datasetMetadata.getLanguage());
         return CompletableFuture.runAsync(() -> {
             try (InputStream input = new URL(url).openStream()) {
                 harvestService.harvest(input, datasetMetadata.getDatasetId(), recordDataEncapsulated,
-                    datasetMetadata.getStepSize());
+                        datasetMetadata.getStepSize(), CompressedFileExtension.ZIP);
```

Comment:
this also needs to be updated :sweat_smile:  I tested with url
`http://ftp.eanadev.org/uploads/sandbox.tar.gz`

Reply:
It should be working now!